drivers/led/lp5569.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2023 Phytec Messtechnik GmbH.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_lp5569
+
+/**
+ * @file
+ * @brief LP5569 LED controller
+ *
+ * The LP5569 is a 9-channel LED driver that communicates over I2C.
+ */
+
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/led.h>
+#include <zephyr/device.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/kernel.h>
+
+#define LOG_LEVEL CONFIG_LED_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(lp5569);
+
+#define LP5569_NUM_LEDS			9
+
+/* General Registers */
+#define LP5569_CONFIG			0x00
+#define LP5569_CHIP_EN			(1 << 6)
+
+#define LP5569_MISC			0x2F
+#define LP5569_POWERSAVE_EN		(1 << 5)
+
+/* PWM base Register for controlling the duty-cycle */
+#define LP5569_LED0_PWM			0x16
+
+struct lp5569_config {
+	struct i2c_dt_spec bus;
+};
+
+static int lp5569_led_set_brightness(const struct device *dev, uint32_t led,
+				     uint8_t brightness)
+{
+	const struct lp5569_config *config = dev->config;
+	uint8_t val;
+	int ret;
+
+	if (led >= LP5569_NUM_LEDS || brightness > 100) {
+		return -EINVAL;
+	}
+
+	/* Map 0-100 % to 0-255 pwm register value */
+	val = brightness * 255 / 100;
+
+
+	ret = i2c_reg_write_byte_dt(&config->bus, LP5569_LED0_PWM + led, val);
+	if (ret) {
```

Comment:
Could you please change it to `if (ret < 0)` everywhere?

Reply:
`i2c_reg_write_byte_dt` returns `i2c_reg_write_byte` which specifies `0` or `-EIO` as return values. Doesn't `ret` cover more cases than `ret < 0`?

Reply:
I did some reading and found that your suggestion seems to be the better approach. When considering safety and extensibility it's better to be explicit and specific. I will change it later today.

Reply:
Thanks, let me know when it will be done.

Reply:
@bbilas just fixed all points.